### PR TITLE
NOTIF-346 Allows warnings on checked checkboxes

### PR DESCRIPTION
**Pull request by josejulio**
Allows to specify a warning message if the checkbox is checked.

Peek 2021-11-22 12-25

### CODE REVIEW

**Comment by josejulio** [comment]
@karelhala could you review? thanks!

File: src/SmartComponents/FormComponents/DescriptiveCheckbox.js [hunk 1]
```diff
@@ -14,16 +15,37 @@ const DescriptiveCheckbox = (props) => {
...
-    input: { onChange, ...input },
+    checkedWarning,
+    input: { onChange, checked, ...input },
   } = useFieldApi({
     ...props,
     type: 'checkbox',
   });
   const formOptions = useFormApi();
+
+  const descriptionProp = useMemo(() => {
```
Comment (karelhala) [inline src/SmartComponents/FormComponents/DescriptiveCheckbox.js:26]: Instead od `memo` I'd use component directly, the rendering is calculated directly by React and we don't have to keep this in memory.

**Review by karelhala — COMMENTED** [review]
Looking good! Just one minor change

File: src/SmartComponents/FormComponents/DescriptiveCheckbox.js (hunk 1 above)
Comment (josejulio) [reply src/SmartComponents/FormComponents/DescriptiveCheckbox.js]: done!

**Comment by josejulio** [comment]
@karelhala Applied suggestion, let me know if we need something else :D

**Review by karelhala — APPROVED** [review]

**Pull request merged**